File: mslibrary-kotlindocs/src/main/java/kr/msleague/message/adapter/CraftItemStackAdapter.java
package kr.msleague.message.adapter;

import kr.msleague.message.api.MSMessageAdapter;
import org.bukkit.craftbukkit.v1_12_R1.inventory.CraftItemStack;

/**
 * MSLibrary 에서 기본적으로 제공하는 CraftItemStack 어댑터입니다.
 */
public class CraftItemStackAdapter implements MSMessageAdapter<CraftItemStack> {
    @Override
    public String reformat(String origin, CraftItemStack obj) {
        return "";
    }
}

File: mslibrary-message/src/main/java/kr/msleague/message/adapter/CraftItemStackAdapter.kt
package kr.msleague.message.adapter

import kr.msleague.message.api.MSMessageAdapter
import kr.msleague.util.locale.friendlyName
import kr.msleague.util.locale.l10nDisplayName
import org.bukkit.craftbukkit.v1_12_R1.inventory.CraftItemStack

class CraftItemStackAdapter : MSMessageAdapter<CraftItemStack> {

    override fun request(obj: CraftItemStack, placeHolder: String): String? {
        return when (placeHolder) {
            "item_name" -> obj.l10nDisplayName
            "item_amount" -> obj.amount.toString()
            "item_type" -> obj.friendlyName
            else -> null
        }
    }
}